### PR TITLE
[DOC] Update style guide and auto documentation

### PR DESCRIPTION
**Pull request by tsalo**
This started as an attempt to update the style guide per @KirstieJane's suggestion in #30, but it ended up sprawling a bit as I tried to get the API documentation working as well. It looks pretty good, because I was able to re-use a lot of code I used in another site, but there are a couple of problems that I don't know how to fix. Namely:
- How do we want to handle the functions that are initialized a level above where they're defined? Not sure that's how to describe what we're doing, but hopefully it makes sense.
- Can we remove the module links in the tables?
- Can we get the `source` links to point to GitHub instead of the code snippets?
 
---

screen shot 2018-05-12 at 8 46 23 pm

---

screen shot 2018-05-12 at 8 46 37 pm

---

screen shot 2018-05-12 at 8 47 15 pm

### CODE REVIEW

**Comment by KirstieJane** [comment]
Yay!! Thank you @tsalo! I’m not knowledgable enough about the details to review the PR, but I do want to say thank you for adding so much to the documentation! So so useful 😍😍😍

**Comment by emdupre** [comment]
This is (another) great PR, @tsalo !!! 

- re documenting only top-level imports: it looks as though we can use the [`..automodule:: pkg` option](https://stackoverflow.com/questions/15115514/how-do-i-document-classes-without-the-module-name). I haven't tried it, personally, but it would be worth a shot !
- You don't like the module links ?? 😢 Or are they not linking out appropriately ?
- To get the source code links I think we'll need to add a sphinx extension&mdash; either [this one](http://www.sphinx-doc.org/en/master/ext/viewcode.html) or [this one](https://github.com/scikit-learn/scikit-learn/blob/master/doc/sphinxext/github_link.py)

**Comment by tsalo** [comment]
@emdupre Unfortunately, I don't think some of those work (or I'm not using them correctly). I'm currently using `..automodule:: pkg` [here](https://github.com/tsalo/tedana/blob/93c38e5ad8e10b327e8d51349331c713b7e48d2d/docs/api.rst), and `sphinx.ext.viewcode` [here](https://github.com/tsalo/tedana/blob/93c38e5ad8e10b327e8d51349331c713b7e48d2d/docs/conf.py#L43), with no luck.

The sklearn extension looks promising, but I have no clue how to use it.

**Comment by emdupre** [comment]
I opened [a PR on your fork](https://github.com/tsalo/tedana/pull/2) with one idea ! I'll tweak more on there to see if we can get this in line with what you're looking for, though it's a great PR as is 😄

**Comment by tsalo** [comment]
Someone else will need to keep working on the Notes in the docstrings, given that I'm not that knowledgeable about the math behind tedana, but other than that I think it's ready to merge.

File: docs/sphinxext/github_link.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from operator import attrgetter
```
Comment (emdupre) [inline docs/sphinxext/github_link.py]: This is totally my oversight, but would you mind adding something here to reference that this was taken from sklearn ? I believe their license is BSD, but I'm sure attribution is appreciated !

Comment (tsalo) [reply docs/sphinxext/github_link.py]: Good catch! Should be good as of 6b1dfb7.

**Comment by emdupre** [comment]
LGTM ! Merging ! 🎉

**Pull request merged**